### PR TITLE
Allow `null` values in `with()`

### PR DESCRIPTION
**Pull request by dajudge**
This used to work before and works with `and()`, so IMO it should also work with `with()`.

### CODE REVIEW

File: src/main/java/com/atomicjar/logging/LogFmtMarker.java [hunk 1]
```diff
@@ -26,13 +26,13 @@ public class LogFmtMarker implements Marker {
...
-    private LogFmtMarker(Map<String, Object> data) {
-        this.data = Collections.unmodifiableMap(data);
+    private LogFmtMarker(Map<String,Object> data) {
+        this.data = data;
     }
 
 
     public static LogFmtMarker with(String key, Object value) {
-        return new LogFmtMarker(Map.of(key, value));
+        return new LogFmtMarker(Collections.emptyMap()).and(key, value);
```
Comment (lanwen) [inline src/main/java/com/atomicjar/logging/LogFmtMarker.java:35]: Looking at the code, null value removes the key from the existing map. But with empty map that doesn't make a lot of sense, so you could simply check if it's null and directly return the marker with an empty map.

**Review by lanwen — CHANGES_REQUESTED** [review]
I think the check is worth implementing

**Comment by dajudge** [comment]
Yeah well I wend for simplicity, but I also really don't have a strong opinion on it X-D

File: src/main/java/com/atomicjar/logging/LogFmtMarker.java (hunk 1 above)
Comment (dajudge) [reply src/main/java/com/atomicjar/logging/LogFmtMarker.java]: done

**Comment by dajudge** [comment]
can you PTAL?

**Review by realjenius — APPROVED** [review]

File: src/main/java/com/atomicjar/logging/LogFmtMarker.java [hunk 2]
```diff
@@ -26,13 +26,14 @@ public class LogFmtMarker implements Marker {
...
-        this.data = Collections.unmodifiableMap(data);
+    private LogFmtMarker(Map<String,Object> data) {
+        this.data = data;
     }
 
 
     public static LogFmtMarker with(String key, Object value) {
-        return new LogFmtMarker(Map.of(key, value));
+        var marker = new LogFmtMarker(Collections.emptyMap());
+        return value == null ? marker : marker.and(key, value);
```
Comment (bsideup) [inline src/main/java/com/atomicjar/logging/LogFmtMarker.java:36]: Shouldn't it remove the old value for the key?

Comment (bsideup) [inline src/main/java/com/atomicjar/logging/LogFmtMarker.java:36]: nit: something like `new LogFmtMarker(value == null ? Map.of() : Map.of(key, value))` would be easier to follow and do less allocations

**Review by bsideup — APPROVED** [review]

File: src/main/java/com/atomicjar/logging/LogFmtMarker.java (hunk 2 above)
Comment (dajudge) [reply src/main/java/com/atomicjar/logging/LogFmtMarker.java]: done

**Comment by dajudge** [comment]
@lanwen any further changes you'd like to see?

**Pull request merged**